src/core/Akka.Persistence/Fsm/PersistentFSMBase.cs

```diff
@@ -332,10 +333,7 @@ public bool IsTimerActive(string name)
         /// <param name="timeout">TBD</param>
         public void SetStateTimeout(TState state, TimeSpan? timeout)
         {
-            if (!_stateTimeouts.ContainsKey(state))
-                _stateTimeouts.Add(state, timeout);
-            else
-                _stateTimeouts[state] = timeout;
+            _stateTimeouts[state] = timeout;
```

Comment:
That was the only method I have seen that was affected from your changes, you don't need to do it for all places that aren't affected by your code